src/packagedcode/plugin_package.py

```diff
@@ -272,7 +278,74 @@ def process_codebase(self, codebase, strip_root=False, package_only=False, **kwa
                 if TRACE_LICENSE and modified:
                     logger_debug(f'packagedcode: process_codebase: add_referenced_license_matches_from_package: modified: {modified}')
 
+@post_scan_impl
+class PackageSummary(PostScanPlugin):
+    """
+    Summary at the Package Level.
+    """
+    run_order = 11
+    sort_order= 11
+
+    options = [
+        PluggableCommandLineOption(('--package-summary',),
+        is_flag=True, default=False,
+        help='Summarize scans by providing License Clarity Score' 
+        'and populating other license/copyright attributes '
+        'for package instances.',
+        required_options=['classify'],
+        help_group=POST_SCAN_GROUP)
+    ]
+
+    def is_enabled(self, package_summary, **kwargs):
+        return package_summary
 
+    def process_codebase(self, codebase, package_summary, **kwargs):
+        """
+        Process the codebase.
+        """
+        if not self.is_enabled(package_summary):
+            return
+        
+        packages = codebase.attributes.packages
+        resource_for_packages = list(get_files_for_packages(codebase))
+        packages_copy = copy.deepcopy(packages) # created a deep copy 
+        package_resources = {}
+        package_attributes_map={}
+        for resource, package_uid in resource_for_packages:
+            if package_uid not in package_resources:
+                package_resources[package_uid] = []
+            package_resources[package_uid].append(resource)
+
+        # Add a 'resources' field to each package in packages_copy
+        for package in packages_copy:
+            package_uid = package['package_uid']
+            if package_uid in package_resources:
+                package['resources'] = [resource.to_dict() for resource in package_resources[package_uid]]
+                
+            scoring_elements, package_attrs= compute_license_score(package, is_codebase=False)
+            license_clarity_score= scoring_elements.to_dict()
+            package_attributes_map[package_uid] = {
+                'license_clarity_score': license_clarity_score,
+                'copyright': package_attrs.copyright,
+                'holder': package_attrs.holder,
+                'notice_text': package_attrs.notice_text,
+                'other_license_expression': package_attrs.other_license_expression,
+                'other_license_expression_spdx': package_attrs.other_license_expression_spdx
+            }
+
+        for package in packages:
+            package_uid = package['package_uid']
+            if package_uid in package_attributes_map:
+                package_attrs = package_attributes_map[package_uid]
+                package['license_clarity_score'] = package_attrs['license_clarity_score']
```

Comment:
The attributes can be stored in list `attributes_to_update` and then looped over and assigned. Cleaner and simpler.

Reply:
Resolved.